client/components/Charts.js

```diff
@@ -370,17 +370,23 @@ function CreateBarChartAll(operations, barchartId) {
 
     // Category -> {Month -> [Amounts]}
     var map = {};
+
+    // Category -> color
+    let colorMap = {};
+
     // Datekey -> Date
     var dateset = {};
-    for (var i = 0; i < operations.length; i++) {
-        var op = operations[i];
-        var c = store.categoryToLabel(op.categoryId);
+    for (let i = 0, size = operations.length; i < size; i++) {
+        let op = operations[i];
+        let c = store.categoryToLabel(op.categoryId);
         map[c] = map[c] || {};
 
         var dk = datekey(op);
         map[c][dk] = map[c][dk] || [];
         map[c][dk].push(op.amount);
         dateset[dk] = +op.date;
+
+        colorMap[c] = colorMap[c] || store.categoryToColor(op.categoryId);
```

Comment:
Now that categories have more than one piece of information, we could change the map in the store, to map from categoryId to category, and use that instead of `categoryToColor` / `categoryToLabel`, so as to prevent a few lookups. Hopefully lookups should be efficient and quick, but I'm afraid this could slow down the entire app. Can be done later, if it's indeed an issue.